inc/classes/subscriber/third-party/plugins/ecommerce/class-woocommerce-subscriber.php

```diff
@@ -418,4 +419,40 @@ public function exclude_product_attributes_cpcss( $excluded_taxonomies ) {
 
 		return array_merge( $excluded_taxonomies, wc_get_attribute_taxonomy_names() );
 	}
+
+	/**
+	 * Set $uid to 0 for certain nonce actions.
+	 *
+	 * @since  3.5.1
+	 * @access public
+	 * @author Soponar Cristina
+	 *
+	 * @param int    $uid    ID of the nonce-owning user.
+	 * @param string $action The nonce action.
+	 *
+	 * @return int $uid      ID of the nonce-owning user.
+	 */
+	public function maybe_revert_uid_for_nonce_actions( $uid, $action ) {
+		if ( $uid && 0 !== $uid && $action && in_array( $action, $this->get_nonce_actions(), true ) ) {
+			$uid = 0;
+		}
+		return $uid;
+	}
+
+	/**
+	 * List with nonce actions which needs to revert the $uid.
+	 *
+	 * @since  3.5.1
+	 * @access private
+	 * @author Soponar Cristina
+	 *
+	 * @return array $nonce_actions List with all nonce actions.
+	 */
+	private function get_nonce_actions() {
+		$nonce_actions = [
+			'wcmd-subscribe-secret', // WooCommerce MailChimp Discount.
+			'td-block', // "Load more" AJAX functionality of the Newspaper theme.
+		];
+		return $nonce_actions;
```

Comment:
you can use the return directly on the array declaration, no need to create a variable here